apitools/base/py/transfer.py

```diff
@@ -632,7 +631,7 @@ def __ConfigureMultipartRequest(self, http_request):
 
     # encode the body: note that we can't use `as_string`, because
     # it plays games with `From ` lines.
-    fp = StringIO.StringIO()
+    fp = six.StringIO()  # email: cStringIO in Py2 / io.StringIO in Py3.
```

Comment:
oh -- so that one's actually an absurd corner case.

the `email` module has an `as_string` function that *should* be what we want to use here. however, by default it assumes you're sending an email (probably not a bad assumption, given the name), and helpfully turns any line starting with 

    From ...

into

    > From ...

we noticed this when some early uploads started picking up two extra bytes on the GCS side (and hence failing consistency checks). probably my favorite bug we hit.